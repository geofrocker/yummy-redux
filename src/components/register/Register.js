import React, { Component } from 'react';
import {connect} from 'react-redux'
import {register} from '../../actions/registerActions'
import toastr from 'toastr'
import {Redirect} from 'react-router-dom'

export class Register extends Component {

    constructor(props){
        super(props);
        this.state={
            userData:{username:'', password:'',name:'',email:'',cpassword:''},
            message:'',
            color:'col-xs-11 alert alert-danger',
            redirect:false
        }
    }

    handleChange = (event) => {
        const field = event.target.name
        let userData = this.state.userData
        userData[field] = event.target.value
        this.setState({ userData })
    }
    regUser = (e) =>{
        e.preventDefault();
        this.setState({
            color:'col-xs-11 alert alert-danger'
        })
        if(this.state.userData.cpassword !== this.state.userData.password){
            return this.setState({
                message: 'Passwords do not match',
                color:'col-xs-11 alert alert-danger'
            })
        }
        this.props.register(this.state.userData).then((response) =>{
            return this.setState(() =>{
                toastr.success('You are now registered')
                return {
                    color:'col-xs-11 alert alert-success',
                    userData:{username:'', password:'',name:'',email:'',cpassword:'',},
                    redirect:true
                }
                
            })
        }).catch(error =>{
            return this.setState(() =>{
                return{
                    color:'col-xs-11 alert alert-danger',
                    message:error.response?error.response.data.Message:'An Error occurred, Try again',
                    userData:{password:'',cpassword:''}
                };
            });
        });
    }
    render(){
        const {userData, message, color,redirect} = this.state
        if(redirect){
            return <Redirect to="/login"/>
        }
        if (localStorage.getItem('isLoggedIn')) {
            return <Redirect to="/dashboard" />;
        }
        return (
        <div className="Register">
            <h1>Register Here</h1>
            {message
                ? <div className={color}>{message}</div>
                : <div></div> 
            }
            <form id="signup-form" onSubmit={this.regUser}>
                <div className="jumbotron col-xs-11">
                    <div className="form-group">
                        <input type="text" className="form-control" placeholder="Name" value={userData.name} onChange={this.handleChange} name="name" id="name" required/>
                    </div>
                    <div className="form-group">
                        <input type="text" className="form-control" placeholder="Username" value={userData.username} onChange={this.handleChange} name="username" id="username" required/>
                    </div>
                    <div className="form-group">
                        <input type="email" className="form-control" placeholder="Email" value={userData.email} onChange={this.handleChange} name="email" id="email" required/>
                    </div>
                    <div className="form-group">
                        <input type="password" className="form-control" placeholder="Password" value={userData.cpassword} onChange={this.handleChange} name="cpassword" id="cpassword" required/>
                    </div>
                    <div className="form-group">
                        <input type="password" className="form-control" placeholder="Comfirm Password" value={userData.password} onChange={this.handleChange} name="password" id="password" required/>
                    </div>
                    <input type="submit" className="btn btn-primary pull-right" value="Register"/>

                    <center>
                        <p>or</p>
                        <p><a href="/login">Already a member? Click here to login</a></p>
                    </center>
                </div>
                
            </form>
            
        </div>
        );
    }
}

function mapStateToProps(state, ownProps){
    return {
      message:state.auth.message,
      status:state.auth.status
    }
}
  
export const mapDispatchToProps = (dispatch) =>({
        register:(data) => dispatch(register(data))    
})
export default connect(mapStateToProps, mapDispatchToProps) (Register);
  