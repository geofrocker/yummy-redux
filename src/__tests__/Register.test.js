import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Register } from '../components/register/Register';

const props = {
  register: () => Promise.resolve(),
};
describe('register component', () => {
  const wrapper = shallow(<Register {...props} />);
  const preventDefault = jest.fn();

  it('renders the Register class', () => {
    expect(wrapper.find('.Register')).toHaveLength(1);
  });

  it('renders five div jsx elements', () => {
    expect(wrapper.find('div')).toHaveLength(8);
  });
  it('renders the sign up form and submit data', () => {
    wrapper.setState({
      name: 'asiimwe', username: 'elizei', email: 'geofrocker2@gmail.com', passwordCandidate: 'psalmsk', password: 'psalms',
    });
    wrapper.find('#signup-form').simulate('submit', { preventDefault });
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(preventDefault).toBeCalled();
  });

  it('render the six inputs', () => {
    expect(wrapper.find('#password')).toHaveLength(1);
    expect(wrapper.find('#cpassword')).toHaveLength(1);
    expect(wrapper.find('#username')).toHaveLength(1);
    expect(wrapper.find('#name')).toHaveLength(1);
    expect(wrapper.find('#email')).toHaveLength(1);
    wrapper.find('#email').simulate('change', { target: { value: 'geofrocker@gmail.com' } });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
