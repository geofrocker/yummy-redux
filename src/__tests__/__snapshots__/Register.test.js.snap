// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register component render the six inputs 1`] = `
<Redirect
  push={false}
  to="/login"
/>
`;

exports[`register component renders the sign up form and submit data 1`] = `
<div
  className="Register"
>
  <h1>
    Register Here
  </h1>
  <div />
  <form
    id="signup-form"
    onSubmit={[Function]}
  >
    <div
      className="jumbotron col-xs-11"
    >
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Name"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="cpassword"
          name="cpassword"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Comfirm Password"
          required={true}
          type="password"
          value=""
        />
      </div>
      <input
        className="btn btn-primary pull-right"
        type="submit"
        value="Register"
      />
      <center>
        <p>
          or
        </p>
        <p>
          <a
            href="/login"
          >
            Already a member? Click here to login
          </a>
        </p>
      </center>
    </div>
  </form>
</div>
`;
